isamples_export_client/duckdb_utilities.py

```diff
@@ -5,6 +5,42 @@
 
 import duckdb
 
+# Specifies the schema of the samples database created when loading the JSON
+SOURCE_COLUMNS = '''{
+sample_identifier: 'VARCHAR',
+"label": 'VARCHAR',
+source_collection: 'VARCHAR',
+has_specimen_category: 'STRUCT("label" VARCHAR)[]',
+has_material_category: 'STRUCT("label" VARCHAR)[]',
+has_context_category: 'STRUCT("label" VARCHAR)[]',
+informal_classification: 'VARCHAR[]',
+keywords: 'STRUCT(keyword VARCHAR)[]',
+produced_by: 'STRUCT(
+    identifier VARCHAR,
+    "label" VARCHAR,
+    description VARCHAR,
+    result_time DATE,
+    sampling_site STRUCT(
+        description VARCHAR,
+        "label" VARCHAR,
+        place_name VARCHAR[],
+        sample_location STRUCT(
+            elevation DOUBLE,
+            latitude DOUBLE,
+            longitude DOUBLE
+        )
+    ),
+    responsibility STRUCT("role" VARCHAR, "name" VARCHAR)[]
+)',
+registrant: 'STRUCT("name" VARCHAR)',
+curation: 'STRUCT(
+    "label" VARCHAR,
+    description VARCHAR,
+    curation_location VARCHAR,
+    access_constraints VARCHAR[]
+)'
+}'''
```

Comment:
This is interesting.  What does it allow `duckdb` to do?

Reply:
I noticed that sometimes DDB doesn't infer the schema consistently when loading content, probably because some optional content isn't present in the records. This just asserts a consistent schema so code using the imported content can access record properties consistently. I think a better option would be to have the parquet include in its metadata some sort of schema model, or perhaps we could include as part of the isamples model.

Reply:
Thanks for the explanation!